src/Freebox.cpp

```diff
@@ -2111,75 +2205,57 @@ PVR_ERROR Freebox::DeleteTimer (const PVR_TIMER & timer, bool force)
 // H O O K S ///////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////////////
 
-int freebox_dialog_select (const vector<long> & v, int selected = -1)
-{
-  // Localize labels.
-  vector<char *> labels;
-  transform (v.begin (), v.end (), back_inserter (labels),
-             [] (long id) {return XBMC->GetLocalizedString (id);});
-
-  // GUI selection.
-  int r = GUI->Dialog_Select (labels[0], (const char **) &(labels[1]), labels.size () - 1, selected);
-
-  // Free localized labels.
-  for (char * label : labels) XBMC->FreeString (label);
-
-  return r;
-}
-
 /* static */
 enum Freebox::Source Freebox::DialogSource (enum Source selected)
 {
-  static const vector<long> LABELS =
+  // Heading.
+  string heading = kodi::GetLocalizedString (PVR_FREEBOX_STRING_CHANNEL_SOURCE);
+
+  // Entries.
+  static const vector<string> LABELS =
   {
-    // Heading.
-    PVR_FREEBOX_STRING_CHANNEL_SOURCE,
-    // Entries.
-    PVR_FREEBOX_STRING_CHANNEL_SOURCE_AUTO,
-    PVR_FREEBOX_STRING_CHANNEL_SOURCE_IPTV,
-    PVR_FREEBOX_STRING_CHANNEL_SOURCE_DVB
+    kodi::GetLocalizedString (PVR_FREEBOX_STRING_CHANNEL_SOURCE_AUTO),
+    kodi::GetLocalizedString (PVR_FREEBOX_STRING_CHANNEL_SOURCE_IPTV),
+    kodi::GetLocalizedString (PVR_FREEBOX_STRING_CHANNEL_SOURCE_DVB)
```

Comment:
About here need to confirmed that works, as it defined in `static` I'm not sure with the calls inside.

Reply:
I haven't built Kodi since I moved to Ubuntu 20.04.
Might take quite some time ;)

Reply:
I successfully built the addon yesterday and it works.
Except it keeps the first language when switching to another one.
I think we should just drop the `static` attribute.

Reply:
Thanks a lot for test!
Yes then makes a static no sense and remove it there, thought not on begin about that, where then only one time called and no more correct if language changed.

Reply:
I make change about, no problem with git :smile: